controller/api/authentication.py

```diff
@@ -9,3 +10,15 @@ def authenticate(self, request):
         Authenticate the request for anyone!
         """
         return AnonymousUser(), None
+
+
+class AnonymousOrAuthenticatedAuthentication(authentication.BaseAuthentication):
+
+    def authenticate(self, request):
+        """
+        Authenticate the request for anyone or if a valid token is provided, a user.
+        """
+        try:
+            return TokenAuthentication.authenticate(TokenAuthentication(), request)
```

Comment:
As far as I can tell, LDAP users are instructed to leave their cluster in `disabled` mode and never register users (The LDAP data given by the LDAP server is translated into a user), so it shouldn't be a problem.